sycl/include/CL/sycl/types.hpp

```diff
@@ -2399,6 +2399,12 @@ struct is_device_copyable<
     T, std::enable_if_t<std::is_trivially_copyable<T>::value>>
     : std::true_type {};
 
+// Specializations of device copyable should propagate onto constants.
+template <typename T>
+struct is_device_copyable<
```

Comment:
@gmlueck
Could you please comment on assuming that if `T` is device copyable then `const T` is device copyable as well? What about `volatile`?

Reply:
Strictly speaking, the SYCL spec does not say that the implementation automatically assumes that "const T" is device copyable if "T" is device copyable.  According to the current wording of the spec, the application is responsible for specializing `is_device_copyable` for all cv-qualified versions of `T` that it wants to declare as "device copyable".

Therefore, it's probably better not to add these lines.  If the goal is to declare that `sycl::ext::oneapi::experimental::properties` is device copyable, it would be better to add specializations just for that type (and any cv-qualified versions we care about).

Reply:
Makes sense to me. I have removed this and specialized `is_device_copyable` for `const properties` instead. We may want to do this for `std::tuple` and `sycl::marray` as well, but I don't think this PR is the right place for that.

Reply:
I am still puzzled by some use-cases where the device copy-ability would depend on CV qualification.
But yeah... keep the door open. :-)